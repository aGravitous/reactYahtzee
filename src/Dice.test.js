import Dice from './Dice';
import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";

let wrapper;

// Smoke test for Dice
it('renders without crashing', () => {
    shallow(<Dice  locked={[false, true, false, false, true]} dice={[1,2,3,4,5]}/>);
});

// // Snapshot test for dice of locked = [false, true, false, false, true]
// // dice = [1,2,3,4,5]
it("matches unlocked snapshot", function (){
    wrapper = mount(<Dice
        locked={[false, true, false, false, true]}
        dice={[1,2,3,4,5]}
        />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});
