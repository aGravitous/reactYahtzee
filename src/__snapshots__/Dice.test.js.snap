// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches unlocked snapshot 1`] = `
<Dice
  dice={
    Array [
      1,
      2,
      3,
      4,
      5,
    ]
  }
  locked={
    Array [
      false,
      true,
      false,
      false,
      true,
    ]
  }
>
  <div
    className="Dice"
  >
    <Die
      idx={0}
      key="0"
      locked={false}
      val={1}
    >
      <button
        className="Die"
        onClick={[Function]}
      >
        1
      </button>
    </Die>
    <Die
      idx={1}
      key="1"
      locked={true}
      val={2}
    >
      <button
        className="Die Die-locked"
        onClick={[Function]}
      >
        2
      </button>
    </Die>
    <Die
      idx={2}
      key="2"
      locked={false}
      val={3}
    >
      <button
        className="Die"
        onClick={[Function]}
      >
        3
      </button>
    </Die>
    <Die
      idx={3}
      key="3"
      locked={false}
      val={4}
    >
      <button
        className="Die"
        onClick={[Function]}
      >
        4
      </button>
    </Die>
    <Die
      idx={4}
      key="4"
      locked={true}
      val={5}
    >
      <button
        className="Die Die-locked"
        onClick={[Function]}
      >
        5
      </button>
    </Die>
  </div>
</Dice>
`;
